### PR TITLE
Reddit: Switch from regex to word trigger

### PR DESCRIPTION
**Pull request by puskin94**
@moollaza I changed the trigger system about this issue: https://github.com/duckduckgo/zeroclickinfo-spice/issues/1591 . Now it works properly but I don't know if this is the best solution

### CODE REVIEW

File: lib/DDG/Spice/RedditSubSearch.pm [hunk 1]
```diff
@@ -16,16 +16,21 @@ attribution twitter => ["https://twitter.com/mithrandiragain","Gary Herreman"],
...
-handle matches => sub {
-	if($1) {
-		return $1;
-	} elsif($2) {
-		return $2;
-	}
-    return;
+handle query_lc => sub {
+
+	$_ =~ s/^(subreddit|r)|(subreddit|r)$//i;
```
Comment (moollaza) [inline lib/DDG/Spice/RedditSubSearch.pm]: Not sure what you're trying to achieve with these replacements. The regex below does all the work for you!

```perl
handle query_lc => sub {
    return unless m/<regex>/);
    my $remainder = $1 || $2;
    return $remainder;
}
```
Something like that should be fine. The regex could possibly do with some improvements, or rather a named capture so we don't have to check for both match groups.

**Comment by puskin94** [comment]
@moollaza I don't know why but I have the power to solve problems with complex solutions, especially if the problem require a simple one

**Comment by puskin94** [comment]
@moollaza Travis failed due not my code:
`The command "dzil authordeps | grep -ve '^\W' | xargs -n 5 -P 10 cpanm  --quiet   --notest  --skip-installed" failed and exited with 123 during .` 
Other tests are passed

**Comment by moollaza** [comment]
LGTM :+1:

**Comment by moollaza** [comment]
Thanks @puskin94

**Pull request merged**